common/workunit/workunit.cpp

```diff
@@ -13540,31 +13540,53 @@ void deleteK8sJob(const char *componentName, const char *job)
     jobname.toLowerCase();
     VStringBuffer deleteJob("kubectl delete job/%s", jobname.str());
     StringBuffer output, error;
-    bool ret = runExternalCommand(output, error, deleteJob.str(), nullptr);
+    bool ret = runExternalCommand(componentName, output, error, deleteJob.str(), nullptr);
     DBGLOG("kubectl delete output: %s", output.str());
     if (error.length())
         DBGLOG("kubectl delete error: %s", error.str());
     if (ret)
         throw makeStringException(0, "Failed to run kubectl delete");
 }
 
-void waitK8sJob(const char *componentName, const char *job, const char *condition)
+void waitK8sJob(const char *componentName, const char *job)
 {
     VStringBuffer jobname("%s-%s", componentName, job);
     jobname.toLowerCase();
+    VStringBuffer waitJob("kubectl get jobs %s -o jsonpath='{.status.active}'", jobname.str());
+    VStringBuffer getPhase("kubectl get pods --selector=job-name=%s --output=jsonpath='{.items[*].status.phase}'", jobname.str());
 
-    if (isEmptyString(condition))
-        condition = "condition=complete";
-
-    // MORE - blocks indefinitely here if you request too many resources
-    VStringBuffer waitJob("kubectl wait --for=%s --timeout=10h job/%s", condition, jobname.str());  // MORE - make timeout configurable
-    StringBuffer output, error;
-    bool ret = runExternalCommand(output, error, waitJob.str(), nullptr);
-    DBGLOG("kubectl wait output: %s", output.str());
-    if (error.length())
-        DBGLOG("kubectl wait error: %s", error.str());
-    if (ret)
-        throw makeStringException(0, "Failed to run kubectl wait");
+    unsigned delay = 100;
+    unsigned start = msTick();
+    unsigned pendingTimeout = 10000;
```

Comment:
I think if the image needs pulling, the pods status remains 'Pending' and can take some time i.e. could be more than 10s

Reply:
What is the conclusion about the 10s delay.  I have seen delays significantly larger than that on azure pulling the first image.

Reply:
I've recoded since Jake's initial comment so that it's not dependent on the "Pending" status but rather on whether the pod has been scheduled.

I need to check whether the "scheduled" happens before the pause for pulling an image...

I probably also need to renaming the "pending" variable